src/python/pants/util/dirutil.py

```diff
@@ -605,6 +606,12 @@ def check_no_overlapping_paths(paths: Sequence[str]) -> None:
         raise ValueError('{} and {} have the same prefix. All paths must be unique and cannot overlap.'.format(path, p))
 
 
+def check_all_relative(*paths: PurePath) -> None:
+  absolute_paths = [p for p in paths if p.is_absolute()]
```

Comment:
Should consider putting this in `__post_init__` for `DirectoryToMaterialize` instead.

Reply:
Great suggestion!